geomesa-gs-monitor-elasticsearch/src/main/scala/org/geomesa/gs/monitor/elastic/ExtendedRequestData.scala

```diff
@@ -0,0 +1,175 @@
+
+  val failed: java.lang.Boolean =
+    Option(requestData.getError)
+      .isDefined
+
+  val timedOut: java.lang.Boolean =
+    Option(requestData.getError)
+      .map(_.getMessage)
+      .contains(TIMEOUT_KEY)
+
+  val bboxCentroid: Point =
+    Option(requestData.getBbox)
+      .map(JTS.toGeometry)
+      .map(_.getCentroid)
+      .orNull
+
+  @transient
+  private val queryFilter: Option[Filter] =
+    Option(requestData.getQueryString)
+      .flatMap(extractCqlFilter)
+
+  val queryAttributes: java.util.List[String] =
+    queryFilter
+      .map(FilterHelper.propertyNames)
+      .asNonEmptyJavaListOrNull
+
+  val queryCentroids: java.util.List[Point] =
+    queryFilter
+      .map(extractFilterCentroids)
+      .asNonEmptyJavaListOrNull
+
+  @transient
+  private val distinguishedName: Option[LdapName] =
+    Option(requestData.getRemoteUser)
+      .flatMap(extractDistinguishedName)
+
+  val commonNames: java.util.List[String] =
+    distinguishedName
+      .map(extractRdnValues(_, "CN"))
+      .asNonEmptyJavaListOrNull
+
+  val organizations: java.util.List[String] =
+    distinguishedName
+      .map(extractRdnValues(_, "O"))
+      .asNonEmptyJavaListOrNull
+
+  val organizationalUnits: java.util.List[String] =
+    distinguishedName
+      .map(extractRdnValues(_, "OU"))
+      .asNonEmptyJavaListOrNull
+}
+
+object ExtendedRequestData extends LazyLogging {
+
+  val CQL_FILTER_START_KEY = "CQL_FILTER="
+  val CQL_FILTER_END_KEY = "&"
+
+  val TIMEOUT_KEY = "TIMEOUT"
+
+  def getGson(excludedFields: Set[String]): Gson = {
+    new GsonBuilder()
+      .registerTypeAdapter(classOf[Point], new JsonSerializer[Point] {
+        private val WKT_WRITER = new WKTWriter
+        override def serialize(point: Point, `type`: Type, context: JsonSerializationContext): JsonElement = {
+          new JsonPrimitive(WKT_WRITER.write(point))
+        }
+      })
+      .registerTypeAdapter(classOf[Date], new JsonSerializer[Date] {
+        override def serialize(date: Date, `type`: Type, context: JsonSerializationContext): JsonElement = {
+          new JsonPrimitive(date.getTime)
+        }
+      })
+      .registerTypeAdapter(classOf[BoundingBox], new JsonSerializer[BoundingBox] {
+        override def serialize(bbox: BoundingBox, `type`: Type, context: JsonSerializationContext): JsonElement = {
+          new JsonPrimitive(bbox.toString)
+        }
+      })
+      .registerTypeAdapter(classOf[Throwable], new JsonSerializer[Throwable] {
+        override def serialize(ex: Throwable, `type`: Type, context: JsonSerializationContext): JsonElement = {
+          new JsonPrimitive(ex.getMessage)
+        }
+      })
+      .registerTypeAdapter(classOf[Array[Byte]], new JsonSerializer[Array[Byte]] {
+        override def serialize(array: Array[Byte], `type`: Type, context: JsonSerializationContext): JsonElement = {
+          new JsonPrimitive(Option(array).map(new String(_, StandardCharsets.UTF_8)).getOrElse(new String))
```

Comment:
might want to base64 encode this, i'm not sure `new String(_)` will work very well

Reply:
So I would do something like: `new String(Base64.encodeBase64(array))`?

Reply:
here's an example: https://github.com/locationtech/geomesa/blob/main/geomesa-convert%2Fgeomesa-convert-common%2Fsrc%2Fmain%2Fscala%2Forg%2Flocationtech%2Fgeomesa%2Fconvert2%2Ftransforms%2FEncodingFunctionFactory.scala#L12-L24